lib/sync/datasetClientsCleaner.js

```diff
@@ -0,0 +1,78 @@
+var util = require('util');
+var parseDuration = require('parse-duration');
+var async = require('async');
+var _ = require('underscore');
+var DatasetClient = require('./DatasetClient');
+var syncUtil = require('./util');
+
+var storage;
```

Comment:
Typically we've found avoiding the `prototype` usage when possible is nice. It avoids errors with `this` not being bound correctly. It looks like this constructor could do so?

```js
function DatasetClientsCleaner () {
  const instance = xtend({}, {
    job: null,
    retentionPeriod: opts.retentionPeriod || '24h',
    checkFrequency: opts.checkFrequency || '1h'
  });

  instance.start = function (immediately, callback) {
    if (immediately) {
      cleanDatasetClients(self.retentionPeriod);
    }
    if (!instance.job) {
      instance.job = setInterval(function() {
        cleanDatasetClients(instance.retentionPeriod);
      }, parseDuration(instance.checkFrequency));
    }
    return cb && cb();
  };

  return instance;
}
```

Obviously depends on how many of these will be created in memory. We also keep functions private similar to how you did so here @wei-lee 👍